src/sentry/workflow_engine/endpoints/utils/test_fire_action.py

```diff
@@ -0,0 +1,37 @@
+import logging
+
+import sentry_sdk
+
+from sentry.shared_integrations.exceptions import IntegrationFormError
+from sentry.workflow_engine.models import Action, Detector
+from sentry.workflow_engine.types import WorkflowEventData
+
+logger = logging.getLogger(__name__)
+
+
+def test_fire_action(
```

Comment:
could testing the trigger live on the `Action` class itself? this just seems like another way of calling trigger, which could be done with `self.trigger` inside `Action`?

e.g.

```python
class Action(Model):

    def test_trigger(event_data, detector) -> list[str]:
        action_exceptions = []
        try:
            self.trigger(
                event_data=event_data,
                detector=detector,
            )
        except Exception as exc:
            ...
```

Reply:
I think it makes sense to live either as a util, or as part of another class. As far as the action is concerned, a trigger is a trigger. If it errors, the caller should know how to handle it gracefully depending on the context.